base/string_utils.hpp

```diff
@@ -584,49 +584,11 @@ int UpperBoundOnChars()
 
   return std::numeric_limits<T>::digits10 + std::is_signed<T>::value + 1;
 }
-
-template <typename T>
-char * to_string_digits(char * buf, T i)
-{
-  do
-  {
-    --buf;
-    *buf = static_cast<char>(i % 10) + '0';
-    i = i / 10;
-  } while (i != 0);
-  return buf;
-}
-
-template <typename T>
-std::string to_string_signed(T i)
-{
-  bool const negative = i < 0;
-  int const sz = UpperBoundOnChars<T>();
-  char buf[sz];
-  char * end = buf + sz;
-  char * beg = to_string_digits(end, negative ? -i : i);
-  if (negative)
-  {
-    --beg;
-    *beg = '-';
-  }
-  return std::string(beg, end - beg);
-}
-
-template <typename T>
-std::string to_string_unsigned(T i)
-{
-  int const sz = UpperBoundOnChars<T>();
-  char buf[sz];
-  char * end = buf + sz;
-  char * beg = to_string_digits(end, i);
-  return std::string(beg, end - beg);
-}
 }  // namespace impl
 
-inline std::string to_string(int32_t i) { return impl::to_string_signed(i); }
-inline std::string to_string(int64_t i) { return impl::to_string_signed(i); }
-inline std::string to_string(uint64_t i) { return impl::to_string_unsigned(i); }
+inline std::string to_string(int32_t i) { return std::to_string(i); }
```

Comment:
1. How fast is std::to_string compared to the previous implementation? Here are some numbers: https://www.zverovich.net/2013/09/07/integer-to-string-conversion-in-cplusplus.html
2. If performance is ok, I would prefer to delete strings::to_string completely and replace it everywhere in the code with std::to_string.

Reply:
> 1. How fast is std::to_string compared to the previous implementation?

Looking at source code it is definitely faster. Current `string_utils.hpp` implementation is just naive one,
the implementation in libstdc+++ and libc++ is similar to Alexandrescu's idea mentioned in your [link](https://www.zverovich.net/2013/09/07/integer-to-string-conversion-in-cplusplus.html). So on graph it should be near `fmt::format_int`

> If performance is ok, I would prefer to delete strings::to_string completely and replace it everywhere in the code with std::to_string

Not sure what do you mean here. I already replaced all occurrences. The `base::to_string` != `std::to_string`. `std::to_string` is implemented for numbers only, the `base::to_string` for any types. I replaced part specialized implemented for numbers, `std::to_string` can replace other part.